### ISSUE TITLE
Does KOReader work on Sony PRS (350, 650, etc.)?

### ISSUE THREAD

**Issue by Darthagnon** [koreader/koreader#12523]
My dad has an old Sony PRS 350 I installed [PRS+](https://www.mobileread.com/forums/showthread.php?t=64510) ([Github archive](https://github.com/natowi/prs-plus)) on a while back. The other ereaders in the family are Kindles with KOReader, and as dad's never been good with tech, it would be helpful if they were all the same. So I was wondering, does KOReader work on Sony PRS ereaders, or is there any way of getting it work?

I did a bit of googling, and all the info I could find was:
- The [Sony PRS-T1](https://www.the-ebook-reader.com/sony-prs-t1.html) seems to [run Android 2.2.1, which is not supported](https://github.com/koreader/koreader/issues/2125)
- The PRS-T1 running Android [was lauded as a change](https://blog.the-ebook-reader.com/2011/10/01/proof-the-sony-reader-wi-fi-runs-android-let-the-rooting-begin/), so I suspect the earlier PRS 350 and 650 etc. models run something different, maybe an obscure Linuxy thing like Kindle?
- [GoodEreader says](https://goodereader.com/blog/electronic-readers/what-you-need-to-know-if-you-purchase-an-old-sony-e-reader) that Kobo provided firmware updates for the Sony ereaders
- [sony-prstux/device.lua](https://github.com/koreader/koreader/blob/master/frontend/device/sony-prstux/device.lua) and [sony-prstux.mk](https://github.com/koreader/koreader/blob/master/make/sony-prstux.mk) files exist in the repo... one looks like a makefile? Probably related to [PRSTUX](https://gitlab.com/phreakuencies/prstux/-/wikis/home), which is a Linux CFW for Sony PRS-Tx ereaders (i.e. the Android-based ones, not the older ones)
- Apparently [KOReader runs on the Sony Digital Paper DPT-RP1](https://github.com/HappyZ/dpt-tools/wiki/How-to-install-KOReader-on-DPT-RP1), though that is a newer device and likely quite different to their older consumer ereaders...

**Comment by pazos** [comment]
No, it will never work on sony android devices below lollipop (android 5, api 21). Kitkat would also work but none of sony devices use that, AFAIK

**Comment by Darthagnon** [comment]
Pretty sure [the PRS series used Linux-based operating system for most models](https://wiki.mobileread.com/wiki/Sony_Portable_Reader) until the PRS-T1 switched to Android. Thank you for confirming, though.

Maybe I can figure out how to use one of the Linux builds on my Dad's PRS 350

**Comment by Frenzie** [comment]
You'd probably have to write an input driver, and if you're unlucky also an eink driver. It's definitely not going to work out of the box. Also it says it's from 2011? So there might be glibc compatibility issues as well.

**Comment by pazos** [comment]
> Pretty sure [the PRS series used Linux-based operating system for most models](https://wiki.mobileread.com/wiki/Sony_Portable_Reader) until the PRS-T1 switched to Android. Thank you for confirming, though.


Yeah, I was talking about android devices. Legacy devices running a heavily stripped down and fairly old version of the linux kernel are too old to work with any modern userspace.

> Maybe I can figure out how to use one of the Linux builds on my Dad's PRS 350

Nope.

You'll need to:

1. exploit a vulnerability to allow code execution as root (maybe already documented?)
2. get root persistence
3. bring a modern userspace (udev won't work below 2.6.32, AFAIR)
4. bring a modern cross toolchain for the target
5. port koreader (what @Frenzie said)

tl;dr: You can use KOReader on a PRS-T1 that was exploited and rooted if you bring your own userspace (PRSTUX). The rest of non-android devices won't be ever supported. They're the **286** of eink devices :)

**Comment by pazos** [comment]
> My dad has an old Sony PRS 350 I installed [PRS+](https://www.mobileread.com/forums/showthread.php?t=64510) ([Github archive](https://github.com/natowi/prs-plus)) on a while back. The other ereaders in the family are Kindles with KOReader, and as dad's never been good with tech, it would be helpful if they were all the same.

In case you consider his next reader I would recommend a **very basic pocketbook***.
They're very good for non-tech seniors :).
- KOReader support
- Integration with main software's home
- Hardware buttons

(*) not a color one, not a high res one. If you go with that aim for a kobo.
